content_scripts/VirtualCamera.js

```diff
@@ -594,6 +351,187 @@ import * as THREE from '../../thirdPartyCode/three/three.module.js';
+                this.threeJsContainer.add(this.followingState.unstabilizedContainer);
+
+                // These boxes could be groups / empty objects rather than meshes, but we have them to help debug
+                let forwardTarget = new THREE.Mesh(new THREE.BoxGeometry(40, 40, 40), new THREE.MeshBasicMaterial({color: '#0000ff'}));
+                forwardTarget.position.set(0, 0, 3000);
+                forwardTarget.name = 'forwardFollowTargetObject';
+                forwardTarget.visible = DISPLAY_PERSPECTIVE_CUBES;
+                this.followingState.unstabilizedContainer.add(forwardTarget);
+                this.followingState.forwardTargetObject = forwardTarget;
+
+                let level = new THREE.Mesh(new THREE.BoxGeometry(40, 40, 40), new THREE.MeshBasicMaterial({color: '#00ffff'}));
+                level.name = 'levelFollowTargetObject';
+                level.visible = DISPLAY_PERSPECTIVE_CUBES;
+                this.threeJsContainer.add(level);
+                this.followingState.levelTargetObject = level;
+            }
+
+            if (!this.followingState.stabilizedContainer) {
+                let container = new THREE.Group();
+                container.name = 'followStabilizedContainer';
+                container.visible = DISPLAY_PERSPECTIVE_CUBES;
+                this.followingState.stabilizedContainer = container;
+                this.threeJsContainer.add(container);
+
+                let obj = new THREE.Mesh(new THREE.BoxGeometry(20, 20, 20), new THREE.MeshBasicMaterial({color: '#ff0000'}));
+                obj.name = 'parametricPositionObject';
+                obj.visible = DISPLAY_PERSPECTIVE_CUBES;
+                container.add(obj);
+
+                let target = new THREE.Mesh(new THREE.BoxGeometry(20, 20, 20), new THREE.MeshBasicMaterial({color: '#ff0000'}));
+                target.name = 'parametricTargetObject';
+                target.visible = DISPLAY_PERSPECTIVE_CUBES;
+                container.add(target);
+
+                // set initial positions of objects otherwise camera following will break
+                this.updateParametricTargetAndPosition(this.followingState.currentFollowingDistance);
+            }
+        }
+        // actually moves the camera to be located at the positionObject and looking at the targetObject
+        // (which are nested inside the stabilized following container)
+        moveCameraToParametricFollowPosition() {
+            let positionObject = realityEditor.gui.threejsScene.getObjectByName('parametricPositionObject');
+            let targetObject = realityEditor.gui.threejsScene.getObjectByName('parametricTargetObject');
+
+            if (positionObject.matrixWorldNeedsUpdate || targetObject.matrixWorldNeedsUpdate) {
+                return; // irrecoverable error in camera position if we continue before Three.js computes the matrixWorld of the new objects
```

Comment:
It would be nice to have  a console.warn here in case it happens frequently

Reply:
added. thanks for the review!